examples/step-49/doc/results.dox

```diff
@@ -225,158 +239,109 @@ void create_3d_grid (Triangulation<3> &triangulation)
 }
 @endcode
 
-With this code, you get a mesh that looks like this:
-
-<img src="https://www.dealii.org/images/steps/developer/step-49.yuhan.2.png" alt="">
-
-The next step is to teach each of the top surfaces that they should be
-curved. We can do this by creating CylinderBoundary objects that
-describe this. A first attempt looks like this:
+This creates the following mesh:
+
+<img src="https://www.dealii.org/images/steps/developer/step-49.yuhan.8.png"
+     alt="" width="400" height="355">
+
+This mesh the right general shape, but the top cells are now polygonal: their
+edges are no longer along circles and we do not have a very accurate
+representation of the original geometry. The next step is to teach the top part
+of the domain that it should be curved. Put another way, all calculations done
+on the top boundary cells should be done in cylindrical coordinates rather than
+Cartesian coordinates. We can do this by creating a CylindricalManifold object
+and associating it with the cells above $y = 3$. This way, when we refine the
+cells on top, we will place new points along concentric circles instead of
+straight lines.
+
+In deal.II we describe all geometries with classes that inherit from
+Manifold. The default geometry is Cartesian and is implemented in the
+FlatManifold class. As the name suggests, Manifold and its inheriting classes
+provide a way to describe curves and curved cells in a general way with ideas
+and terminology from differential geometry: for example, CylindricalManifold
+inherits from ChartManifold, which is describes a geometry through pull backs
+and push forwards. In general, one should think that the Triangulation class
+describes the topology of a domain (in addition, of course, to storing the
+locations of the vertices) while the Manifold classes describe the geometry of a
+domain (e.g., whether or not a pair of vertices lie along a circular arc or a
+straight line). A Triangulation will refine cells by doing computations with the
+Manifold associated with that cell regardless of whether or not the cell is on
+the boundary. Put another way: the Manifold classes do not need any information
+about where the boundary of the Triangulation actually is: it is up to the
+Triangulation to query the right Manifold for calculations on a cell. Most
+Manifold functions (e.g., Manifold::get_intermediate_point) know nothing about
+the domain itself and just assume that the points given to it lie along a
+geodesic. In this case, with the CylindricalManifold constructed below, the
+geodesics are arcs along circles orthogonal to the $z$-axis centered along the
+line $(0, 3, z)$.
+
+Since all three top parts of the domain use the same geodesics, we will
+mark all cells with centers above the $y = 3$ line as being cylindrical in
+nature:
 
 @code
+  const Tensor<1, 3> axis {{0.0, 0.0, 1.0}};
+  const Point<3> axial_point {0, 3.0, 0.0};
```

Comment:
The different number of curly brackets looks inconsistent.

Reply:
You're right. Unfortunately, in C++11 we need two pairs for the Tensor. I changed the Point constructor to be a function call.

I don't know how else to get two `const` objects here.